pandas/tests/arithmetic/test_numeric.py

```diff
@@ -1116,45 +1116,75 @@ def test_ufunc_compat(self, holder, dtype):
         tm.assert_equal(result, expected)
 
     # TODO: add more dtypes
-    @pytest.mark.parametrize("dtype", [np.int64, np.uint64, np.float64])
+    @pytest.mark.parametrize("dtype", [np.int64, np.uint64, np.float64, np.complex128])
     def test_ufunc_coercions(self, index_or_series, dtype):
         idx = index_or_series([1, 2, 3, 4, 5], dtype=dtype, name="x")
         box = index_or_series
 
         result = np.sqrt(idx)
-        assert result.dtype == "f8" and isinstance(result, box)
-        exp = Index(np.sqrt(np.array([1, 2, 3, 4, 5], dtype=np.float64)), name="x")
+        if result.dtype.kind == "c":
+            assert result.dtype == dtype and isinstance(result, box)
+            exp_dtype = dtype
+        else:
+            assert result.dtype == "f8" and isinstance(result, box)
+            exp_dtype = np.float64
+        exp = Index(np.sqrt(np.array([1, 2, 3, 4, 5], dtype=exp_dtype)), name="x")
         exp = tm.box_expected(exp, box)
         tm.assert_equal(result, exp)
 
         result = np.divide(idx, 2.0)
-        assert result.dtype == "f8" and isinstance(result, box)
-        exp = Index([0.5, 1.0, 1.5, 2.0, 2.5], dtype=np.float64, name="x")
+        if result.dtype.kind == "c":
+            assert result.dtype == dtype and isinstance(result, box)
+            exp_dtype = dtype
+        else:
+            assert result.dtype == "f8" and isinstance(result, box)
+            exp_dtype = np.float64
+        exp = Index([0.5, 1.0, 1.5, 2.0, 2.5], dtype=exp_dtype, name="x")
         exp = tm.box_expected(exp, box)
         tm.assert_equal(result, exp)
 
         # _evaluate_numeric_binop
         result = idx + 2.0
-        assert result.dtype == "f8" and isinstance(result, box)
-        exp = Index([3.0, 4.0, 5.0, 6.0, 7.0], dtype=np.float64, name="x")
+        if result.dtype.kind == "c":
+            assert result.dtype == dtype and isinstance(result, box)
+            exp_dtype = dtype
+        else:
+            assert result.dtype == "f8" and isinstance(result, box)
+            exp_dtype = np.float64
```

Comment:
it looks like a bunch of these assertions are rendered redundant by the tm.assert_equal calls below them.  could we get away with just setting exp_dtype?  could even write a helper to de-duplicate these

Reply:
I think that would work.  I'll give it a try.